### ISSUE TITLE
Biome encountered an unexpected error

### ISSUE THREAD

**Issue by ziyak97** [biomejs/biome#10806]
### Environment information

```
biome rage
CLI:
  Version:                      2.2.5
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.6.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Error:                        Found an unknown key `preset`.
  Error:                        Found an unknown key `tailwindDirectives`.
  Status:                       Loaded with errors
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0

```

### What happened?

```
bun format
$ biome check --write --unsafe && biome format --write
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_rowan/src/ast/mod.rs:207:13
Thread Name: biome::workspace_worker_4
Message: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_rowan/src/ast/mod.rs:207:13
Thread Name: biome::workspace_worker_2
Message: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_rowan/src/ast/mod.rs:207:13
Thread Name: biome::workspace_worker_3
Message: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_rowan/src/ast/mod.rs:207:13
Thread Name: biome::workspace_worker_3
Message: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_rowan/src/ast/mod.rs:207:13
Thread Name: biome::workspace_worker_1
Message: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_rowan/src/ast/mod.rs:207:13
Thread Name: biome::workspace_worker_9
Message: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_rowan/src/ast/mod.rs:207:13
Thread Name: biome::workspace_worker_8
Message: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_rowan/src/ast/mod.rs:207:13
Thread Name: biome::workspace_worker_8
Message: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
Stack Trace: Re-run with `RUST_BACKTRACE=1` to capture the stack trace

src/components/posts/activity-templates/previews/animal-flow-display-preview.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

src/components/posts/activity-templates/previews/bouldering-display-preview.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

src/components/posts/activity-templates/previews/hot-yoga-display-preview.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

src/components/posts/activity-templates/previews/nutrition-display-preview.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

src/components/posts/activity-templates/previews/scuba-diving-display-preview.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

src/components/posts/activity-templates/previews/snorkeling-display-preview.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

src/components/posts/activity-templates/previews/stretching-display-preview.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

src/components/posts/activity-templates/previews/trekking-display-preview.tsx internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: Tried to cast node with kind JSX_TAG_EXPRESSION as `"biome_js_syntax::generated::nodes::AnyJsxAttributeValue"` but was unable to cast
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

Checked 2079 files in 2s. No fixes applied.
Formatted 2086 files in 784ms. No fixes applied.
```

I am sharing one of the files here. 

```
"use client";

import { motion } from "framer-motion";
import { cn } from "@/lib/utils";
import { ActivityScenePreview } from "./reference-preview-primitives";

export type SnorkelingDisplayPreviewProps = {
  participants: Array<{
    id: string;
    displayName: string;
    avatarUrl: string | null;
    username: string | null;
  }>;
};

const Snorkel = ({ className }: { className?: string }) => (
  <div className={cn("relative h-[45px] w-[20px]", className)}>
    {/* Tube */}
    <div
      className="absolute right-0 top-0 bottom-[10px] w-[5px] rounded-t-full bg-cyan-400"
      style={{
        boxShadow:
          "inset 1px 1px 2px rgba(255,255,255,0.4), 0 3px 6px rgba(0,0,0,0.3)",
      }}
    >
      {/* Top bright orange splash guard */}
      <div className="absolute top-0 left-0 right-0 h-[8px] rounded-t-full bg-orange-500" />
    </div>
    {/* Mouthpiece curve */}
    <div className="absolute bottom-0 right-[2px] h-[15px] w-[15px] rounded-b-full border-b-[5px] border-l-[5px] border-neutral-600 bg-transparent" />
  </div>
);

export function SnorkelingDisplayPreview({
  participants,
}: SnorkelingDisplayPreviewProps) {
  return (
    <ActivityScenePreview
      label="Snorkeling"
      backgroundClassName="bg-[#005f73]"
      participants={participants}
      participantClassName="translate-y-5"
      participantGlowClassName={"bg-[#94d2bd]/25 blur-[22px]"}
      scene={
        <>
          {/* Shimmering turquoise water gradient */}
          <div className="absolute inset-0 bg-[linear-gradient(180deg,#0a9396_0%,#005f73_50%,#002b3d_100%)]" />

          {/* Shimmering sunlight rays */}
          <div
            className="absolute inset-0 opacity-25"
            style={{
              background:
                "repeating-linear-gradient(60deg, transparent, transparent 30px, rgba(238,245,219,0.15) 50px, rgba(238,245,219,0.15) 70px)",
            }}
          />

          {/* Small bubbles */}
          {[
            { left: "20%", bottom: "10%", size: 3, delay: 0.5 },
            { left: "80%", bottom: "20%", size: 4, delay: 1.8 },
            { left: "45%", bottom: "15%", size: 5, delay: 0.2 },
          ].map((bubble, i) => (
            <motion.div
              key={`bubble-${i}`}
              animate={{
                y: [0, -80],
                x: [0, Math.sin(i) * 3, 0],
                opacity: [0, 0.6, 0],
              }}
              transition={{
                duration: 5 + i * 2,
                repeat: Infinity,
                ease: "easeOut",
                delay: bubble.delay,
              }}
              className={
                "absolute rounded-full border border-white/35 bg-white/10"
              }
              style={{
                left: bubble.left,
                bottom: bubble.bottom,
                width: bubble.size,
                height: bubble.size,
              }}
            />
          ))}

          {/* Shallow reef floor silhouette */}
          <div
            className="absolute bottom-0 right-0 left-0 h-[24%] bg-[#001c26]"
            style={{
              clipPath:
                "polygon(0% 100%, 0% 50%, 15% 40%, 35% 65%, 60% 45%, 80% 70%, 100% 50%, 100% 100%)",
            }}
          />

          {/* Floating Snorkel */}
          <motion.div
            animate={{ y: [0, -3, 0], rotate: [0, 5, 0] }}
            transition={{ duration: 6, repeat: Infinity, ease: "easeInOut" }}
            className="absolute bottom-[16%] right-[12%] z-10 drop-shadow-[0_8px_16px_rgba(0,0,0,0.4)]"
          >
            <Snorkel />
          </motion.div>
        </>
      }
      overlays={
        <>
          <div className="pointer-events-none absolute inset-0 bg-[radial-gradient(circle_at_50%_15%,rgba(148,210,189,0.3)_0%,transparent_40%),linear-gradient(180deg,transparent_0%,rgba(0,0,0,0.15)_100%)]" />
        </>
      }
    />
  );
}
```
The issue is this fragment. Once i remove the fragment it works.

```
   overlays={
        <>
          <div className="pointer-events-none absolute inset-0 bg-[radial-gradient(circle_at_50%_15%,rgba(148,210,189,0.3)_0%,transparent_40%),linear-gradient(180deg,transparent_0%,rgba(0,0,0,0.15)_100%)]" />
        </>
      }
```

### Expected result

This needs to work.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Please check if this is an issue **with the latest**

**Comment by biome-cookie** [comment]
Hello @ziyak97, please provide a minimal reproduction. You can use one of the following options:

- Provide a link to [our playground](https://biomejs.dev/playground), if it's applicable.
- Provide a link to GitHub repository. To easily create a reproduction, you can use our interactive CLI via `npm create @biomejs/biome-reproduction`

Issues marked with `S-Needs repro` will be **closed** if they have **no activity within 3 days**.

**Comment by ziyak97** [comment]
@ematipico what kinda useless AI is this? It's asking to do all the work it should be doing after i have spoon fed it the repro. Should it not do the task it's asking me to do itself?

my biome version ->     "@biomejs/biome": "2.5.1"

**Comment by ematipico** [comment]
- It's not an AI, it's only an automated message
- Your version **isn't** 2.5.1. Your environment information that you pasted show `2.2.5` with errors in the configuration file
- We're asking to provide **us** a way to reproduce the issue, for example a minimal repository. Without it, we can't triage the problem. With the info we have now, we can't reproduce it.

**Comment by ziyak97** [comment]
Got it thought it was an AI (maybe it should have been?). Will repro it later today. 

I am slightly confused with the versioning. 

biome version
CLI:        2.2.5


Here it clearly says 2.2.5, but my package.json & actual bin in node_modules both point to 2.5.1

bin in node_modules

```
{
  "name": "@biomejs/biome",
  "version": "2.5.1",
  "bin": {
    "biome": "bin/biome"
  },
  "homepage": "https://biomejs.dev",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/biomejs/biome.git",
    "directory": "packages/@biomejs/biome"
  },
  "author": "Emanuele Stoppa",
  "license": "MIT OR Apache-2.0",
  "bugs": "https://github.com/biomejs/biome/issues",
  "description": "Biome is a toolchain for the web: formatter, linter and more",
  "files": [
    "bin/biome",
    "configuration_schema.json",
    "README.md",
    "LICENSE-APACHE",
    "LICENSE-MIT",
    "ROME-LICENSE-MIT"
  ],
  "keywords": [
    "format",
    "lint",
    "toolchain",
    "JavaScript",
    "TypeScript",
    "JSON",
    "JSONC",
    "JSX",
    "TSX",
    "CSS",
    "GraphQL"
  ],
  "engines": {
    "node": ">=14.21.3"
  },
  "funding": {
    "type": "opencollective",
    "url": "https://opencollective.com/biome"
  },
  "publishConfig": {
    "provenance": true
  },
  "optionalDependencies": {
    "@biomejs/cli-win32-x64": "2.5.1",
    "@biomejs/cli-win32-arm64": "2.5.1",
    "@biomejs/cli-darwin-x64": "2.5.1",
    "@biomejs/cli-darwin-arm64": "2.5.1",
    "@biomejs/cli-linux-x64": "2.5.1",
    "@biomejs/cli-linux-arm64": "2.5.1",
    "@biomejs/cli-linux-x64-musl": "2.5.1",
    "@biomejs/cli-linux-arm64-musl": "2.5.1"
  }
}
```

Is there some separate cli updates i need to follow?

**Comment by ematipico** [comment]
That's why we're asking you to create a fresh new repository, as suggested by the automated message. Let us know once you have created one.

**Comment by ziyak97** [comment]
Repro - https://github.com/ziyak97/biome-repro-1782816611575/tree/main

**Comment by dyc3** [comment]
Thank you, my clanker was able to reproduce this

**Comment by notorious94** [comment]
@dyc3 any possibility of fixing this soon?
